pandas/core/generic.py

```diff
@@ -10091,8 +10091,8 @@ def truncate(
         if isinstance(ax, MultiIndex):
             setattr(result, self._get_axis_name(axis), ax.truncate(before, after))
 
-        if copy:
-            result = result.copy()
+        if copy or copy is None:
+            result = result.copy(deep=copy)
```

Comment:
This might not be needed, since the `result` is from an indexing operation, and already should be a lazy copy (when using a slice) with refs set up

Reply:
We could rewrite as ``if copy or copy is None and _using_copy_on_write()``, but we need the check for the non cow case

Reply:
Should it be `if copy or (copy is None and not _using_copy_on_write())`? (so with brackets and when _not_ using CoW)

Essentially we need to same behaviour as this pattern that is used elsewhere, I think:

```
        if copy is None:
            if _using_copy_on_write():
                copy = False
            else:
                copy = True
        if copy:
            result = result.copy()

Reply:
Ah yes, forgot the not